### PR TITLE
Improved ISP chooser logic

### PR DESCRIPTION
**Pull request by ryanrolds**
I cleaned up the validation logic for the ISP chooser, fixing the data not showing up after emptying the ISP list.

### CODE REVIEW

File: app/assets/javascripts/mapbox.coffee [hunk 1]
```diff
@@ -387,12 +387,14 @@ update_all_option = (elem) ->
   return if $.inArray(elem.prop('id'), ['provider', 'stats_provider', 'zip_code', 'census_code']) < 0
   selected_elem_value = $("#selected_#{elem.prop('id')}").val().split(',')
 
-  if $.inArray('all', selected_elem_value) >= 0
+  new_selected = elem.val()
+
+  if ($.inArray('all', selected_elem_value) != -1 ||
+     $.inArray('all', new_selected) != -1) && new_selected != null
```
Comment (Kourchenko) [inline app/assets/javascripts/mapbox.coffee]: Pure semantics, but I keep chained logic statements on one line.
  if ( $.inArray('all', selected_elem_value) != -1 || $.inArray('all', new_selected) != -1 ) && 
    new_selected != null

OR we could structure it a bit more
if new_selected != null && 
    ( $.inArray('all', selected_elem_value) != -1 ||
    $.inArray('all', new_selected) != -1 )

Comment (ryanrolds) [reply app/assets/javascripts/mapbox.coffee]: Is the code actually wrong? This looks like a stylistic objection.

Comment (ryanrolds) [reply app/assets/javascripts/mapbox.coffee]: If I don't wrap at the `||` the line is 91 characters long, which is passed my target of ~80.

Comment (Kourchenko) [reply app/assets/javascripts/mapbox.coffee]: Mostly stylistic objection. I suggest putting the "new_selected != null" logic statement first for readability.
if new_selected != null &&
( $.inArray('all', selected_elem_value) != -1 ||
$.inArray('all', new_selected) != -1 )

**Pull request merged**